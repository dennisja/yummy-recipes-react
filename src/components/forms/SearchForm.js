import React, {Component} from 'react';
import PropTypes from 'prop-types';

class SearchForm extends Component{
    static initialState= {
        q: "",
    };

    state = SearchForm.initialState;

    handleSearchInputSubmit = (event)=>{
        event.preventDefault();
        this.props.getSearchInput(this.state);
    }

    handleSearchInputChange = (event)=>{
        const {name, value} = event.target;
        this.setState({
            [name]:value,
        });
    }

    getStateOnKeyup = event=>{
        this.props.getSearchInput(this.state);
    }
    

    render(){
        const {q} = this.state;
        const { handleSearchInputBlur, handleSearchInputFocus, onMouseEnter, onMouseLeave} = this.props;
        
        return(
            <form onSubmit={this.handleSearchInputSubmit} id="searchForm">
                <div className ="row"  onMouseLeave={onMouseLeave} onMouseEnter={onMouseEnter}>
                    <div className="input-field col s9 m8 offset-m1 ">
                        <label htmlFor="q">What are you looking for?</label>
                        <input 
                            type="search" 
                            name="q" value={q} id="q" 
                            onChange={this.handleSearchInputChange}
                            onFocus={handleSearchInputFocus}
                            onBlur={handleSearchInputBlur}
                            onKeyUp={this.getStateOnKeyup}/>
                    </div>
                    <div className="input-field col s2 m2">
                        <button className="btn btn-small orange white-text" type="submit"><i className="fa fa-search" /></button>
                    </div>
                </div>
            </form>
        )
    }
}

SearchForm.propTypes = {
    onMouseEnter: PropTypes.func,
    onMouseLeave: PropTypes.func,
    handleSearchInputBlur: PropTypes.func,
    handleSearchInputFocus: PropTypes.func,
}
export default SearchForm;