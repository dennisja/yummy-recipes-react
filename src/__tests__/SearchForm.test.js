import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import SearchForm from '../components/forms/SearchForm';

describe('Test SearchForm', () => {
    const formWrapper = shallow(<SearchForm
      onMouseEnter={jest.fn()}
      onMouseLeave={jest.fn()}
      handleSearchInputBlur={jest.fn()}
      handleSearchInputFocus={jest.fn()}
      getSearchInput={jest.fn()}
    />);
    const preventDefault = jest.fn();


    it('renders properly', () => {
        expect(shallowToJson(formWrapper)).toMatchSnapshot();
    });

    it('renders a form element', () => {
        expect(formWrapper.find('form')).toHaveLength(1);
    });

    it('renders the correct form fields', () => {
        expect(formWrapper.find('#q').length).toBe(1);
    });

    it('submits data', () => {
        formWrapper.find('form').simulate('submit', { preventDefault });
        expect(preventDefault).toBeCalled();
    });
});
