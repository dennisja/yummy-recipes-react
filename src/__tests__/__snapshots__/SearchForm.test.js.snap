// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchForm renders properly 1`] = `
<form
  id="searchForm"
  onSubmit={[Function]}
>
  <div
    className="row"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="input-field col s9 m8 offset-m1 "
    >
      <label
        htmlFor="q"
      >
        What are you looking for?
      </label>
      <input
        id="q"
        name="q"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyUp={[Function]}
        type="search"
        value=""
      />
    </div>
    <div
      className="input-field col s2 m2"
    >
      <button
        className="btn btn-small orange white-text"
        type="submit"
      >
        <i
          className="fa fa-search"
        />
      </button>
    </div>
  </div>
</form>
`;
